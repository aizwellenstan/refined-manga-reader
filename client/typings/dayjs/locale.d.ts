declare module 'dayjs/locale/*' {
  export default any;
}
