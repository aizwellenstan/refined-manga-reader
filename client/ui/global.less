html,
body,
#root {
    height: 100%;
    width: 100%;
}

html,
body {
    margin: 0;
    padding: 0;
}
.main-container {
    display: flex;
    height: 100%;
    width: 100%;

    > .main-search-container {
        margin: 20vh auto 0;
    }
}

.home-search-option {
    color: inherit;
    display: flex;
    flex-flow: row nowrap;
    font-size: 1rem;
    height: 100%;
    width: 100%;

    & > &-title {
        flex: 1;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    & > &-status {
        font-size: 0.7em;
        margin-left: auto;
    }
}

// HACKS

.ant-tolltip-inner {
    font-size: 1rem;
}
