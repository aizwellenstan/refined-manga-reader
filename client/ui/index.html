<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Refined itsukara.link</title>
  <meta name="description" content="âœ¨ itsukara.link with refined interface!">
  <link rel="icon" href="/favicon.ico">
  <link rel="manifest" href="/manifest.json">
  <meta name="theme-color" content="#f80652">
  <link rel="apple-touch-icon" href="/apple-touch-icon.png">
  <link rel="mask-icon" href="/mask-icon.svg" color="#f80652">
  <meta name="apple-mobile-web-app-title" content="Refined itsukara.link">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta property="og:title" content="Refined Itsukara.link">
  <meta property="og:description" content="âœ¨ itsukara.link with refined interface!">
  <meta property="og:url" content="https://refined-itsukara-link.neet.love">
  <meta property="og:image" content="https://refined-itsukara-link.neet.love/screenshot.png" />
  <meta name="twitter:card" content="summary_large_image">
</head>
<body>
  <div id="root" role="application"></div>
</body>
</html>
