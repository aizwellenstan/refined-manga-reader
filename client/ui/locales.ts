import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import localizedFormat from 'dayjs/plugin/localizedFormat';
import 'dayjs/locale/ja';
import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import en from 'client/locales/en/translation.json';
import ja from 'client/locales/ja/translation.json';

export function normalizeLanguageForDayjs(lng: string) {
  return [
    'de-at',
    'en-gb',
    'es-do',
    'es-us',
    'pt-br',
    'sr-cyrl',
    'zh-cn',
    'zh-hk',
    'zh-tw',
  ].includes(lng)
    ? lng
    : lng.split('-')[0];
}

export const getLocale = () => {
  dayjs.extend(localizedFormat);
  dayjs.extend(relativeTime);
  dayjs.locale('en');

  i18next.on('initialized', options => {
    if (!options.lng) return;
    dayjs.locale(normalizeLanguageForDayjs(options.lng));
  });

  i18next.on('languageChanged', lng => {
    if (!lng) return;
    dayjs.locale(normalizeLanguageForDayjs(lng));
  });

  i18next.use(LanguageDetector).init({
    debug: process.env.NODE_ENV === 'development',
    resources: {
      en: { translation: en },
      ja: { translation: ja },
    },
    ns: ['translation'],
    defaultNS: 'translation',
    fallbackLng: 'en',
    react: {
      wait: true,
    },
    interpolation: {
      escapeValue: true,
    },
  });

  return i18next;
};
