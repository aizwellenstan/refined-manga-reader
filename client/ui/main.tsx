import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { StoreContext } from 'redux-react-hook';
import { Root } from './views/root';
import { configureStore } from './redux/store';
import { BrowserRouter } from 'react-router-dom';
import * as OfflinePluginRuntime from 'offline-plugin/runtime';

(async () => {
  if (process.env.NODE_ENV === 'production') {
    OfflinePluginRuntime.install();
  }

  const mountNode = document.getElementById('root');
  if (!mountNode) return;

  const store = configureStore();

  ReactDOM.render(
    <StoreContext.Provider value={store}>
      <BrowserRouter>
        <Root />
      </BrowserRouter>
    </StoreContext.Provider>,
    mountNode,
  );
})();
