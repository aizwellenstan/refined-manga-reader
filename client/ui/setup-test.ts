import { GlobalWithFetchMock } from 'jest-fetch-mock';
import 'jest-styled-components';
import { getLocale } from './locales';

getLocale();

const customGlobal: GlobalWithFetchMock = global as GlobalWithFetchMock;
customGlobal.fetch = require('jest-fetch-mock');
customGlobal.fetchMock = customGlobal.fetch;

// React-i18next
// See: https://react.i18next.com/misc/testing
jest.mock('react-i18next', () => ({
  withNamespaces: () => (Component: React.ComponentType) => {
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  },
}));
