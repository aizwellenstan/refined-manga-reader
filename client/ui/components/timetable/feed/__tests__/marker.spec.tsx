import React from 'react';
import { cleanup, render } from 'react-testing-library';
import { withTheme } from 'client/ui/test-utils';
import { Marker } from '../marker';
import dayjs from 'dayjs';

const event = {
  id: 1069,
  name: "神ゲーらしい【The Beginner's Guide】",
  description:
    'にじさんじ所属バーチャルライバー剣持刀也 防御力：ちくわの700倍 【Twitter】https://twitter.com/rei_Toya_rei',
  public: 1,
  url: 'https://www.youtube.com/watch?v=nm5VvO9Z4gY',
  start_date: '2019-02-28T00:00:00.000+09:00',
  end_date: '2019-02-28T01:00:00.000+09:00',
  recommend: false,
  liver: {
    id: 23,
    name: '剣持刀也',
    avatar:
      'https://s3-ap-northeast-1.amazonaws.com/liver-icons/400x400/Kenmochi_Touya.png',
    color: '#a590af',
  },
};

describe('<Marker />', () => {
  afterEach(cleanup);

  it('renders marker with given props', () => {
    const Component = withTheme(Marker);
    const result = render(
      <Component
        event={event}
        row={0}
        startDate={dayjs('2019-02-28T00:00:00.000+09:00')}
      />,
    );

    expect(result.container.firstChild).toMatchSnapshot();
  });
});
