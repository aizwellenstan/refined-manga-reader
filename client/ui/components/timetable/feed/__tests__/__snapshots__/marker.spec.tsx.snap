// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Marker /> renders marker with given props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 115px;
  left: 0;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 6px;
  border-radius: 99px;
  color: #ffffff;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  width: 40px;
  height: 40px;
  margin-right: 4px;
  border-radius: 50%;
  background-color: #ffffff;
}

.c2 {
  min-width: 0;
}

.c3 {
  display: block;
  margin: 0;
  overflow: hidden;
  font-size: 14px;
  font-weight: bold;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c4 {
  display: block;
  overflow: hidden;
  opacity: 0.8;
  font-size: 12px;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<a
  class="c0"
  href="https://www.youtube.com/watch?v=nm5VvO9Z4gY"
  rel="noreferrer"
  style="width: 282px; transform: translate(9px, 0px); background-color: rgb(165, 144, 175); box-shadow: 0 2px 6px #a590af;"
  tabindex="0"
  target="__blank"
  title="神ゲーらしい【The Beginner's Guide】"
>
  <img
    class="c1"
    src="https://s3-ap-northeast-1.amazonaws.com/liver-icons/400x400/Kenmochi_Touya.png"
  />
  <div
    class="c2"
  >
    <h4
      class="c3"
    >
      神ゲーらしい【The Beginner's Guide】
    </h4>
    <span
      class="c4"
    >
      剣持刀也
    </span>
  </div>
</a>
`;
